tests/unit/-private/better-errors-test.js

```diff
@@ -0,0 +1,57 @@
+import Ember from 'ember';
+import { test, module } from 'qunit';
+import { create } from 'ember-cli-page-object';
+import {
+  throwBetterError
+} from 'ember-cli-page-object/-private/better-errors';
+
+const page = create({
+  foo: {
+    scope: '.foo',
+    bar: {
+      scope: '.bar',
+      focus() {}
+    }
+  }
+});
+
+module('Unit | throwBetterError');
+
+test('shows the expected error message when `selector` is not passed in', function(assert) {
+  assert.expect(1);
+
+  const fn = () => {
+    throwBetterError(page.foo.bar, 'focus', 'Oops!');
+  };
+  const expectedError = new Ember.Error(
+    "Oops!\n\nPageObject: 'page.foo.bar.focus'"
+  );
+
+  assert.throws(fn, expectedError, 'should show message & property path');
+});
+
+test('shows the expected error message when `selector` is passed in', function(assert) {
+  assert.expect(1);
+
+  const fn = () => {
+    throwBetterError(page.foo.bar, 'focus', 'Oops!', { selector: '.foo .bar' });
+  };
+  const expectedError = new Ember.Error(
+    "Oops!\n\nPageObject: 'page.foo.bar.focus'\n  Selector: '.foo .bar'"
+  );
+
+  assert.throws(fn, expectedError, 'should show message, property path, & selector');
+});
+
+test('logs the error to the console', function(assert) {
+  assert.expect(2);
+
+  const origEmberLoggerError = Ember.Logger.error;
+
+  Ember.Logger.error = (msg) => {
+    assert.equal(msg, "Oops!\n\nPageObject: 'page.foo.bar.focus'");
+    Ember.Logger.error = origEmberLoggerError;
```

Comment:
Ah, I see...in case the test fails and line 53 is never hit? Are you thinking something like this?
```javascript
Ember.Logger.error = (msg) => {
  try {
    assert.equal(msg, "Oops!\n\nPageObject: 'page.foo.bar.focus'");
  } finally {
    Ember.Logger.error = origEmberLoggerError;
  }
};
```